gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/impl/sql/GenericStatement.java

```diff
@@ -133,6 +133,8 @@ Licensed to the Apache Software Foundation (ASF) under one or more
 
         public static final Pattern SKIP_CANCEL_STMTS = Pattern.compile(
             "^\\s*(DROP|TRUNCATE)\\s+(TABLE|INDEX)\\s+", Pattern.CASE_INSENSITIVE);
+	public static final Pattern DELETE_STMT = Pattern.compile(
+			"^\\s*DELETE\\s+FROM\\s+.*", Pattern.CASE_INSENSITIVE);
         private static final Pattern ignoreStmts = Pattern.compile(
```

Comment:
Need to keep in mind that all these regexes are broken if there are comments in middle for example. Is it not simple enough to find it out from the QueryTreeNode instead which should be available here? If not, then we need to track that these should come from parser and not by a simple regex match.